src/main/java/ch/njol/skript/effects/base/SetEffect.java

```diff
@@ -0,0 +1,147 @@
+				"set %" + type + "%'[s] " + property + " to %boolean%");
+	}
+	
+	/**
+	 * Registers an effect with patterns "set property of %type% to %boolean%", "set %types%'[s] property to %boolean%"
+	 * and "make %types% makeProperty" with "force %types% to makeProperty"
+	 * 
+	 * @param effect The SetEffect class that you are registering.
+	 * @param property The property of the syntax.
+	 * @param makeProperty The property of the syntax used for "make %types% makeProperty".
+	 * @param type The type classinfo of the syntax. Can be plural.
+	 */
+	public static void registerMake(Class<? extends SetEffect<?>> effect, String property, String makeProperty, String type) {
+		Skript.registerEffect(effect, "set " + property + " of %" + type + "% to %boolean%",
+				"set %" + type + "%'[s] " + property + " to %boolean%",
+				"make %" + type + "% " + makeProperty,
+				"force %" + type + "% to " + makeProperty,
+				"make %" + type + "% not " + makeProperty,
+				"force %" + type + "% to not " + makeProperty);
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public final boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		make = matchedPattern >= 2;
+		if (exprs.length == 1) {
+			if (!make)
+				throw new SkriptAPIException("There was not two expressions in the SetEffect class '" + getClass().getName() + "' exprs: " + Arrays.toString(exprs));
+			expression = (Expression<T>) exprs[0];
+			negated = matchedPattern >= 4;
+			return true;
+		}
+		expression = (Expression<T>) exprs[matchedPattern];
+		value = (Expression<Boolean>) exprs[matchedPattern ^ 1];
+		return true;
+	}
+
+	/**
+	 * Returns the value of the boolean expression used.
+	 * 
+	 * @param event The event that is calling this syntax.
+	 * @return boolean from the boolean expression used in this set effect.
+	 */
+	@Nullable
+	protected final Boolean getBoolean(Event event) {
+		if (value == null)
+			return null;
+		return value.getSingle(event);
+	}
+
+	protected final boolean isNegated() {
+		return negated;
+	}
+
+	/**
+	 * Returns the expression that was registered as the type for this SetEffect.
+	 * 
+	 * @return Expression<T> for getting the values of the expression used.
+	 */
+	protected final Expression<T> getExpression() {
+		return expression;
+	}
+
+	/**
+	 * Return a BiComsumer that will be used to apply the boolean value to the type.
+	 */
+	protected abstract BiConsumer<T, Boolean> apply();
+
+	/**
+	 * Return the property name of this SetEffect used for the {@link #toString(Event, boolean)} method.
+	 * 
+	 * @return String representing the name of the property registered for this syntax.
+	 */
+	protected abstract String getPropertyName();
+
+	@Override
+	protected void execute(Event event) {
+		boolean value = make ? !negated : negated ? !getBoolean(event) : getBoolean(event);
+		getExpression().stream(event).forEach(expression -> apply().accept(expression, value));
```

Comment:
we can probably "cache" the value of the apply() call - if we're going to have the user return a biconsumer it seems pointless to make a call to obtain it multiple times

although im a little unsure of using a biconsumer at all - why not just a method like `execute(Event, boolean)`?